src/Functions/ReplaceRegexpImpl.h

```diff
@@ -271,17 +274,24 @@ struct ReplaceRegexpImpl
 
         for (size_t i = 0; i < input_rows_count; ++i)
         {
-            size_t hs_from = i > 0 ? haystack_offsets[i - 1] : 0;
+            size_t hs_from = haystack_offsets[i - 1];
             const char * hs_data = reinterpret_cast<const char *>(haystack_data.data() + hs_from);
             const size_t hs_length = static_cast<unsigned>(haystack_offsets[i] - hs_from - 1);
 
-            size_t ndl_from = i > 0 ? needle_offsets[i - 1] : 0;
+            size_t ndl_from = needle_offsets[i - 1];
             const char * ndl_data = reinterpret_cast<const char *>(needle_data.data() + ndl_from);
             const size_t ndl_length = static_cast<unsigned>(needle_offsets[i] - ndl_from - 1);
             std::string_view needle(ndl_data, ndl_length);
 
             if (needle.empty())
-                throw Exception(ErrorCodes::ARGUMENT_OUT_OF_BOUND, "Length of the pattern argument in function {} must be greater than 0.", name);
+            {
+                res_data.insert(res_data.end(), hs_data, hs_data + hs_length);
+                res_data.push_back(0);
+
+                res_offsets[i] = res_offsets[i - 1] + hs_length + 1;
+                res_offset = res_offsets[i];
```

Comment:
The following code is equivalent but shorter and more consistent with how we assign `res_offsets[i]` if the needle isn't empty:
```
res_offset += hs_length + 1;
res_offsets[i] = res_offset;
```